Automattic/jetpack
# Masterbar: Update user options only when they change

Fixes https://github.com/Automattic/jetpack/issues/18949

#### Changes proposed in this Pull Request:

Due to [this bug in Core](https://core.trac.wordpress.org/ticket/22192), `update_metadata` performs an unnecessary `UPDATE` SQL query because is unable to check correctly whether a `false` metadata was already `false` in the database.

To work around that, in this PR we're changing the values of the user options needed by nav unification so they are stored as `0`/`1` strings which don't present that problem, slightly improving the performance.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Go to https://wordpress.com/me/account.
- Make sure you're using a LTR language like "English".
- Make sure the "Show advanced pages" toggle is off.
- Install Jetpack Beta on an Atomic site.
- Go to `/wp-admin/admin.php?page=jetpack-beta`
- Switch to `Bleeding Edge` to confirm the issue first.
- Click on "Debug" in the toolbar and then on "Queries" in the sidebar navigation.
- Note how there is a couple of `UPDATE` SQL queries than runs on every page load for updating the `jetpack_wpcom_is_rtl` and `jetpack_admin_menu_link_destination` user options. <img width="1021" alt="Screen Shot 2021-04-13 at 13 43 55" src="https://user-images.githubusercontent.com/1233880/114548403-ffa8a280-9c5f-11eb-9c75-2391d7ea1bfa.png">
- Close the debug panel.
- Switch to the branch of this PR.
- Debug the SQL queries again.
- You should see again a couple of SQL queries updating the `jetpack_wpcom_is_rtl` and `jetpack_admin_menu_link_destination` user options (this is expected, because previously we were storing an empty value and now we are storing a `0` value). <img width="1011" alt="Screen Shot 2021-04-13 at 17 40 46" src="https://user-images.githubusercontent.com/1233880/114580842-802acb80-9c7f-11eb-9e01-f8ef2904c007.png">
- Reload the page and debug the SQL queries once again.
- Make sure there isn't any SQL query updating the the `jetpack_wpcom_is_rtl` and `jetpack_admin_menu_link_destination` user option.
